lib/assert.js

```diff
@@ -230,6 +237,10 @@ function looseDeepEqual(actual, expected) {
   if (isRegExp(actual) && isRegExp(expected)) {
     return areSimilarRegExps(actual, expected);
   }
+  if (actual instanceof Error && expected instanceof Error) {
+    if (actual.message !== expected.message || actual.name !== expected.name)
```

Comment:
Could you compare `code` as well?

Reply:
`code` will be compared if it is set by the user as a non enumerable property. This is only intended for the non enumerable properties `message` and `name` from the default `Error` classes and not for userland sub-classing.

Reply:
Ahh Ok. The `keys` of `expected` are the constraint.

Reply:
This would fail because the left prototype is not the same as the right one.